### PR TITLE
improve epp metrics

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- rename epp metrics to follow the pattern: eventting_epp-<backend>-<metric>
- update dashboards to use renamed metrics
- introduce metric for total requests to epp

**Related issue(s)**

### CODE REVIEW

File: components/event-publisher-proxy/pkg/metrics/collector.go [hunk 1]
```diff
@@ -2,32 +2,53 @@ package metrics
...
+	BackendLatencyKey = "eventing_epp_backend_duration_milliseconds"
+	// backendLatencyHelp help text for the backendLatencyHelp metric
+	backendLatencyHelp = "The duration of sending events to the messaging server in milliseconds"
+
+	// BackendRequestsKey name of the eventRequests metric
+	BackendRequestsKey = "eventing_epp_backend_requests_total"
+	// backendRequestsHelp help text for event backendRequests metric
+	backendRequestsHelp = "The total number of backend requests"
+
+	// durationKey name of the durationMetric metric
```
Comment (friedrichwilken) [inline components/event-publisher-proxy/pkg/metrics/collector.go:31]:
```suggestion
	// durationKey name of the duration metric
```

**Comment by raypinto** [comment]
Not a review but a heads up: We need to update the docs: https://kyma-project.io/docs/kyma/latest/04-operation-guides/operations/evnt-02-eventing-metrics/

File: components/event-publisher-proxy/pkg/metrics/collector.go [hunk 2]
```diff
@@ -123,6 +169,20 @@ func (c *Collector) RecordEventType(eventType, eventSource string, statusCode in
 }
 
 // RecordRequests records an eventRequests metric
-func (c *Collector) RecordRequests(statusCode int, destSvc string) {
-	c.requests.WithLabelValues(fmt.Sprint(statusCode), destSvc).Inc()
+func (c *Collector) RecordBackendRequests(statusCode int, destSvc string) {
+	c.backendRequests.WithLabelValues(fmt.Sprint(statusCode), destSvc).Inc()
+}
+
+func (c *Collector) MetricsMiddleware(destService string) mux.MiddlewareFunc {
```
Comment (nachtmaar) [inline components/event-publisher-proxy/pkg/metrics/collector.go:176]: nit: Can you add a comment to the method ? Describe what it does ?

**Review by nachtmaar — COMMENTED** [review]
LGTM. Only few nits. Found also one text occurrence of the old metric name:

```
pkg/metrics/metricstest/metricstest.go
15:// EnsureMetricErrors ensures metric eventing_epp_errors_total exists.
```

**Review by franpog859 — COMMENTED** [review]
Hmm, I think something has gone wrong while resolving some conflicts. I see some changes that shouldn't be needed here, like changes in Dockerfile from for example Compass Runtime Agent. Do you really need them? And therefore, do you need my review here, @k15r ?

**Comment by k15r** [comment]
> Hmm, I think something has gone wrong while resolving some conflicts. I see some changes that shouldn't be needed here, like changes in Dockerfile from for example Compass Runtime Agent. Do you really need them? And therefore, do you need my review here, @k15r ?

no clue what happened here... i rebased the pr and now everything looks normal again

**Review by nachtmaar — APPROVED** [review]

File: components/event-publisher-proxy/pkg/handler/handler.go [hunk 3]
```diff
@@ -208,12 +209,12 @@ func (h *Handler) sendEventAndRecordMetrics(ctx context.Context, event *cev2even
 	result, err := h.Sender.Send(ctx, event)
 	duration := time.Since(start)
 	if err != nil {
-		h.collector.RecordError()
+		h.collector.RecordBackendError()
```
Comment (vvxxvvxx) [inline components/event-publisher-proxy/pkg/handler/handler.go:212]: Hi Korbinian, can we collect both `RecordBackendLatency` and `RecordBackendRequests` here and assign an http response code (5xx or 4xx) to them? as we also need the latency for the request with the error response code. and for the request number of `RecordBackendRequests`, it should also count the requests number with error response code.

**Pull request merged**